### ISSUE TITLE
Error to install php on ubuntu 14.04

### ISSUE THREAD

**Issue by noamtahan** [phpbrew/phpbrew#249]
Hi,
I need to use php 5.3.XX on apache for example 5.4.0.
I tried:

```
phpbrew install 5.4.0 +default
```

```
noam@noam-Inspiron-5423:~$ phpbrew install 5.4.0 +default
===> Downloading from http://museum.php.net/php5/php-5.4.0.tar.bz2
--2014-06-16 16:00:10--  http://museum.php.net/php5/php-5.4.0.tar.bz2
Resolving museum.php.net (museum.php.net)... 108.163.197.98
Connecting to museum.php.net (museum.php.net)|108.163.197.98|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 11439508 (11M) [application/x-bzip2]
Saving to: ‘php-5.4.0.tar.bz2’

100%[=======================================================================================================>] 11,439,508  1.96MB/s   in 14s    

2014-06-16 16:00:24 (818 KB/s) - ‘php-5.4.0.tar.bz2’ saved [11439508/11439508]

===> php-5.4.0.tar.bz2 downloaded.
===> Extracting php-5.4.0.tar.bz2...
===> Applying patch file for freetype include path bug...
patching file configure
Hunk #1 succeeded at 38802 (offset -411 lines).
patch unexpectedly ends in middle of line
Hunk #2 succeeded at 40984 with fuzz 1 (offset -411 lines).
===> Configuring 5.4.0...


Use tail command to see what's going on:
   $ tail -f /home/noam/.phpbrew/build/php-5.4.0/build.log


configure: WARNING: bison versions supported for regeneration of the Zend/PHP parsers: 1.28 1.35 1.75 1.875 2.0 2.1 2.2 2.3 2.4 2.4.1 2.4.2 2.4.3 2.5 (found: 3.0.2).
configure: WARNING: bison versions supported for regeneration of the Zend/PHP parsers: 1.28 1.35 1.75 1.875 2.0 2.1 2.2 2.3 2.4 2.4.1 2.4.2 2.4.3 2.5 (found: none).
===> Building...
/home/noam/.phpbrew/build/php-5.4.0/ext/dom/node.c: In function ‘dom_canonicalization’:
/home/noam/.phpbrew/build/php-5.4.0/ext/dom/node.c:1898:21: error: dereferencing pointer to incomplete type
    ret = buf->buffer->use;
                     ^
In file included from /home/noam/.phpbrew/build/php-5.4.0/main/php.h:38:0,
                 from /home/noam/.phpbrew/build/php-5.4.0/ext/dom/node.c:26:
/home/noam/.phpbrew/build/php-5.4.0/ext/dom/node.c:1900:40: error: dereferencing pointer to incomplete type
     RETVAL_STRINGL((char *) buf->buffer->content, ret, 1);
                                        ^
/home/noam/.phpbrew/build/php-5.4.0/Zend/zend_API.h:571:20: note: in definition of macro ‘ZVAL_STRINGL’
   const char *__s=(s); int __l=l;   \
                    ^
/home/noam/.phpbrew/build/php-5.4.0/ext/dom/node.c:1900:5: note: in expansion of macro ‘RETVAL_STRINGL’
     RETVAL_STRINGL((char *) buf->buffer->content, ret, 1);
     ^
make: *** [ext/dom/node.lo] Error 1
Errornoam@noam-Inspiron-5423:~$ 
```

I added the lines:

```
export PHPBREW_SET_PROMPT=1
source /home/noam/.phpbrew/bashrc
```

at the end of the file `~/.bashrc`
I tried to:

```
noam@noam-Inspiron-5423:~$ sudo apt-get install -y libicu48
[sudo] password for noam: 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package libicu48 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'libicu48' has no installation candidate
noam@noam-Inspiron-5423:~$ 
```

Maybe this is really necessary?
Thanks

**Comment by jagandecapri** [comment]
@noahwilliamsson Thx. Building php 5.3.10 in Ubuntu 14.04 works now for me. Maybe include this in the Troubleshooting section. https://github.com/phpbrew/phpbrew/wiki/TroubleShooting

Thanks again!!

**Comment by dan-lee** [comment]
:+1: Thanks @noahwilliamsson! I added some information about this in the troubleshooting section.

**Comment by markseu** [comment]
:+1: Thanks @noahwilliamsson, building PHP 5.4.4 on Ubuntu 14.04 works now.

**Comment by Aranmobinarann** [comment]
Hi
This compilation error in ext/dom/node.c at line 1898 (ret = buf->buffer->use;) is caused by a conflict with the C++ keyword use when building with modern GCC (4.8+).

The issue is fixed in later PHP 5.4.x releases (e.g., 5.4.45). I recommend trying:

```bash
phpbrew install 5.4.45 +default
```

If you specifically need PHP 5.4.0, you can apply a patch (search for "PHP 5.4.0 use keyword fix") or compile with an older GCC version (e.g., GCC 4.4).

Reference: This bug was reported and resolved in PHP's official release after 5.4.0.
